### PR TITLE
feat(error): added error_handler for schedule

### LINKED ISSUE #83: [FEATURE] API specs and tests for incorrect input for /schedule
## Description

Add API specs, error handlers and tests for incorrect input.

## Describe the solution you'd like

 I guess an exception will be thrown at some point if we specify a wrong date for example 2021-42-42. It would be great if we handle that exception nicely in our API. Flask has a functionalyty to set handlers for exceptions. In that case we can reply to the caller with a valid json with error description and something like bad Request status

### PR DESCRIPTION
**Pull request by cornhubius**
**What**

- added error_handler
- added test_error_hendler

closed #83

### CODE REVIEW

File: backend/tests/test_error_hendler.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import server as app
+import pytest
+from tools.create_db_for_tests import create_test_app_with_db
+
+
+@pytest.mark.parametrize("date", [("/schedule/week=2021-15-25&filter=booking"),
+                                  ("/schedule/week=2021-05-40&filter=booking"),
+                                  ("/schedule/day=2021-15-25&filter=booking"),
+                                  ("/schedule/day=2021-05-40&filter=available")])
+def test_status_500(date):
```
Comment (mentortechabc) [inline backend/tests/test_error_hendler.py:10]: probably name is inconsistent with the error code

**Review by mentortechabc — APPROVED** [review]
Looks good!

**Comment by mentortechabc** [comment]
There is a typo in "hAndler" but we will fix it later :)

**Review by DramaTourG — APPROVED** [review]

File: backend/tests/test_error_hendler.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    response = test_app.get(date)
+    assert response.status == '400 BAD REQUEST'
+
+
+@pytest.mark.parametrize("date", [("/schedule/week=2021-10-25&filter=booking"),
+                                  ("/schedule/week=2021-05-10&filter=available"),
+                                  ("/schedule/day=2021-10-25&filter=booking"),
+                                  ("/schedule/day=2021-10-25&filter=available")])
+def test_status_200(date):
+    create_test_app_with_db()
```
Comment (DramaTourG) [inline backend/tests/test_error_hendler.py:22]: this function is superfluous

Comment (cornhubius) [reply backend/tests/test_error_hendler.py:22]: really)
the function did not work before, but I thought that it was trying to get data from the local database, but it is different for everyone :)

**Pull request merged**